File: Swift/04-Swift基本语法/04-OC调用Swift/04-OC调用Swift/ViewController.m
//
//  ViewController.m
//  04-OC调用Swift


#import "ViewController.h"
#import "SwiftModule-Swift.h"

// 在Build Settings 中搜索 Product Module Name
// 然后把里面的内容修改为SwiftModule
// 哪里需要使用Swift中的类，就倒入#import "SwiftModule-Swift.h"就可以使用了

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    SXTWeiBoModel *mode = [[SXTWeiBoModel alloc] init];
    mode.name = @"小欢";
    mode.icon = @"icon.png";
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift/04-Swift基本语法/05-命名空间/05-命名空间/ViewController.swift
//
//  ViewController.swift
//  05-命名空间


import UIKit

/*
namespace默认是项目名称，同一个命名空间全局共享

修改namespace
-> Build Settings中搜索product name,即可修改
注意：新改的名字不能含有“中文”, 不能以“数字”开头,不能有“-”
*/

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // 通过一个字符串来创建一个类(ViewController)
        // 1.动态的获取命名空间的名字
        let str: String = NSBundle.mainBundle().infoDictionary!["CFBundleExecutable"] as! String

        // 2.通过字符串得到一个AnyClass
        let cls: AnyClass = NSClassFromString(str + "." + "ViewController")!
        
        // 3.获取控制器的class
        let clsVC = cls as! UIViewController.Type
        
        // 4.通过控制器的class来实例化对象
        let viewVC = clsVC.init()
        
        print("viewVC: \(viewVC)")
        print("self: \(self)")
        
        let model = Model()
        print(model)
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

